pkg/instancetype/preference/apply/device.go

```diff
@@ -0,0 +1,104 @@
+//nolint:gocyclo
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+package apply
+
+import (
+	virtv1 "kubevirt.io/api/core/v1"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+)
+
+func ApplyDevicePreferences(preferenceSpec *v1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
```

Comment:
Why is this func exported again?

Reply:
https://github.com/kubevirt/kubevirt/blob/a83db2fb18d646cb37901ffe3f448864a2c991e2/pkg/virt-controller/watch/vm/vm.go#L3328-L3339

I'll rework this when dropping `InstancetypeMethods`, likely breaking device preferences out into their own handler that we can provide.